### DISCUSSION TITLE
Add filter to data models page

### DISCUSSION THREAD

**Discussion by pierre-H** [directus/directus#5990] [Feature Requests]
In the `/admin/settings/data-model`, it would be great to have search bar to filter the display of data models.
This would help in editing a model when you have many of them.

**Comment by philleepflorence** [comment]
I can see the value in locating a collection from a very long list of collections in the data model settings...
Lemme see if my note is relevant.

### Current Options
- The _Toggle Filter_ in the [Page Sidebar](https://docs.directus.io/concepts/application/).  
- The _Header Sort Function_ in the [Table Layout](https://docs.directus.io/concepts/layouts/#table-layout) .

### Some Thoughts
Additional features and user actions for context; assuming a mean deviation of 50 - 200 collections.
- Add some more options (display collection translation) for the Sidebar Filter for average projects. 
    - e.g. _tap and tap to filter_ vs _tap and keystrokes to filter_. 
-  Emphasize the _Header Sort Function_ for average projects. 
    - e.g. _tap and scroll to filter_ vs _tap and keystrokes to filter._ 
- Use the `<search-input>` in [Collection Module](https://github.com/directus/directus/blob/main/app/src/modules/collections/routes/collection.vue) in [Data Model Module](https://github.com/directus/directus/blob/main/app/src/modules/settings/routes/data-model/collections/collections.vue) for larger projects or searching via translations, et al.
    -  e.g. _collections displays translation_ and _data model displays name_.
- Allow [Layout Extensions](https://docs.directus.io/concepts/layouts/) access to Settings Routes for the niche use cases. 

### Visual Guide
A lot of clients really liked and used this image to make UI requests!
![Directus-Layout](https://user-images.githubusercontent.com/2633551/120462079-947a7300-c368-11eb-8e00-a3b0d8f1f460.png)

> I wonder what option is feasible for most user needs or use cases.
> _Translation and name confusion were from a few instances._

@pierre-H what number of collections were you thinking or any additional thoughts?

**Reply by pierre-H** [reply]
> @pierre-H what number of collections were you thinking or any additional thoughts?

Your notes are interesting !
For me, the search should be always proposed, whatever the number of models.

**Reply by philleepflorence** [reply]
I don't object to a search input personally...just considering different use cases...current solutions...filter, search and toggle actions...what field is searched, et al

**Comment by benhaynes** [comment]
I think the standard "search" action button in the header would work here. I think it should either have focus on page load, or at least have a single keyboard shortcut to give it focus.

**Reply by philleepflorence** [reply]
That would fit the layout and many expectations..._Start typing to search or scroll to continue_...

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
